integration_tests/steps/balancing_market.py

```diff
@@ -168,15 +168,17 @@ def follow_device_registry_energy_rate_or(context, device1, device2):
     house1 = next(filter(lambda x: x.name == "House 1", context.simulation.area.children))
     house2 = next(filter(lambda x: x.name == "House 2", context.simulation.area.children))
 
-    assert all(int(round(trade.offer.price / trade.offer.energy, 2)) in negative_energy_rate
-               for device in [house1, house2, context.simulation.area]
-               for market in device.past_balancing_markets
-               for trade in market.trades if trade.offer.energy < 0)
-
-    assert all(int(round(trade.offer.price / trade.offer.energy, 2)) in positive_energy_rate
-               for device in [house1, house2, context.simulation.area]
-               for market in device.past_balancing_markets
-               for trade in market.trades if trade.offer.energy > 0)
+    assert all(
+        int(round(trade.offer_bid.price / trade.offer_bid.energy, 2)) in negative_energy_rate
```

Comment:
This is a good opportunity to use the `Offer/Bid.energy_rate` member and replace the division. This applies to also other occurrences in the PR

Reply:
This has exposed a bug, atm in two_sided's accept_bid we do not update the energy_rate of the bid.
I have fixed it